### PR TITLE
Added section about ConfigurationBuilder

### LINKED ISSUE #34869: Configuration docs all assume you want to run a web service
**The Microsoft.Extensions.Configuration* docs all assume you want to run a web service**

The .net (note: not ASP.NET) documentation at https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration shows readers how to create a hosted web service, not how to instantiate a configuration builder.

```using IHost host = Host.CreateDefaultBuilder(args).Build();```

This also means readers need to add Microsoft.Extensions.Hosting to their project etc. It's missed the point of explaining how to actually use the Configuration extensions. 

.NET 6.0+

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: b458a209-339d-e896-4e85-6d1c6a7aa1f9
* Version Independent ID: 0d5de37c-8f89-b6a5-3132-2c5214196bfd
* Content: [Configuration - .NET](https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration)
* Content Source: [docs/core/extensions/configuration.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/configuration.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Added section about `ConfigurationBuilder`

Fixes #34869

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/extensions/configuration.md](https://github.com/dotnet/docs/blob/b9a53eb666108048b6501e710458cd7be3239d36/docs/core/extensions/configuration.md) | [Configuration in .NET](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/configuration?branch=pr-en-us-36328) |

### CODE REVIEW

File: docs/core/extensions/configuration.md [hunk 1]
```diff
@@ -32,7 +32,36 @@ Given one or more configuration sources, the <xref:Microsoft.Extensions.Configur
 
 ## Configure console apps
 
-.NET console applications created using the [dotnet new](../tools/dotnet-new.md) command template or Visual Studio by default *do not* expose configuration capabilities. To add configuration in a new .NET console application, [add a package reference](../tools/dotnet-add-package.md) to [Microsoft.Extensions.Hosting](https://www.nuget.org/packages/Microsoft.Extensions.Hosting). Modify the *Program.cs* file to match the following code:
+.NET console applications created using the [dotnet new](../tools/dotnet-new.md) command template or Visual Studio by default *don't* expose configuration capabilities. To add configuration in a new .NET console application, [add a reference](../tools/dotnet-add-package.md) to [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration). This package is the foundation for configuration in .NET apps. It provides the <xref:Microsoft.Extensions.Configuration.ConfigurationBuilder> and related types.
```
Comment (adegeo) [inline docs/core/extensions/configuration.md:35]: nit: Personally I prefer telling them to add a the XYZ NuGet package. At this point if they're adding configuration to an app they hopefully know what NuGet is. Of course, you could still rearrange the sentence to tell them to do so via the command line with a link to add-package or via Visual Studio with a link to an article about how to to do that.

File: docs/core/extensions/configuration.md [hunk 2]
```diff
@@ -32,7 +32,36 @@ Given one or more configuration sources, the <xref:Microsoft.Extensions.Configur
...
+    {
+        ["SomeKey"] = "SomeValue"
+    })
+    .Build();
+
+Console.WriteLine(configuration["SomeKey"]);
+
+// Outputs:
+//   SomeValue
+```
```
Comment (adegeo) [inline docs/core/extensions/configuration.md:51]: Why no snippet?

EDIT: Further, I think this example is kind of useless, sorry! 😁  If it's for configuration I would assume the most common scenario would be consuming the _appsettings.json_ file. I took this from my [modernizing a .net framework app after porting to .net article](https://learn.microsoft.com/en-us/dotnet/core/porting/modernize).

```csharp
using Microsoft.Extensions.Configuration;

internal class Program
{
    internal static IConfiguration Config { get; private set; }

    private static void Main(string[] args)
    {
        Config = new ConfigurationBuilder()
            .AddJsonFile("appsettings.json")
            .Build();

        // Use the config file to get a connection string
        string? myConnectionString = Config.GetConnectionString("database");

        // Run the rest of your app
    }
}
```

Then you could also show the appsettings.json file too.

**Review by adegeo — APPROVED** [review]
Approved with some things for you to consider.

File: docs/core/extensions/configuration.md (hunk 1 above)
Comment (IEvangelist) [inline docs/core/extensions/configuration.md:35]:
```suggestion
.NET console applications created using the [dotnet new](../tools/dotnet-new.md) command template or Visual Studio by default *don't* expose configuration capabilities. To add configuration in a new .NET console application, [add a package reference](../tools/dotnet-add-package.md) to [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration). This package is the foundation for configuration in .NET apps. It provides the <xref:Microsoft.Extensions.Configuration.ConfigurationBuilder> and related types.
```

File: docs/core/extensions/configuration.md (hunk 2 above)
Comment (IEvangelist) [reply docs/core/extensions/configuration.md:51]: This makes sense in the context (and flow) of the article. I'll add it as a snippet.

**Pull request merged**